### PR TITLE
Update CI workflow for 'Local Flow' scenario

### PR DESCRIPTION
**Pull request by ivysochyn**
# Update CI workflow for 'Local Flow' scenario

This PR updates the local flow build process and the CI workflow to improve compatibility with the latest `bazel-orfs` changes.

This PR modifies the following:
* `.github/scripts/build_local_target.sh`
  * Modified flow to operate on 'in-place' stages (e.g. `do-floorplan` instead of `floorplan`)
  * Modified the script to handle additional sub-stages in the synthesis (`synth`) stage, including `do-yosys-canonicalize` and `do-yosys`
  * Updated logging to provide more detailed and specific output for each build and run stage

* `.github/workflows/ci.yml`
  * Updated the Docker image used in the local flow to `openroad/orfs:v3.0-1114-g46acc762` to align with the one used by the project. Resolves the incompatible database schema revision error
  * Removed an outdated conditional statement in the CI workflow that was no longer necessary

### CODE REVIEW

**Review by oharboe — APPROVED** [review]

**Comment by ivysochyn** [comment]
@oharboe Here's the fix for the CI as discussed earlier today.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -73,7 +73,7 @@ jobs:
     name: Local flow - test targets
     runs-on: ubuntu-22.04
     container:
-      image: openroad/orfs:v3.0-1022-g69767b6a@sha256:adc74e5343a7638919f747eb74cbc67b5b71b18d9012608fef912ccee534c458
+      image: openroad/orfs:v3.0-1114-g46acc762@sha256:ae4df23391c26bcc48a506f8e0d0da73742d1b6cb3b1dc02f4d5ea71170195b5
```
Comment (oharboe) [inline .github/workflows/ci.yml:76]: Add comment here and in MODULE.bazel to keep them in sync

**Review by oharboe — CHANGES_REQUESTED** [review]
nit

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -73,7 +73,8 @@ jobs:
     name: Local flow - test targets
     runs-on: ubuntu-22.04
     container:
-      image: openroad/orfs:v3.0-1022-g69767b6a@sha256:adc74e5343a7638919f747eb74cbc67b5b71b18d9012608fef912ccee534c458
+      # NOTE: Ensure this is kept in sync with MODULE.bazel
```
Comment (oharboe) [inline .github/workflows/ci.yml:76]: actually, can we make this an error if it is different?

**Review by oharboe — CHANGES_REQUESTED** [review]
Nit

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ivysochyn) [reply .github/workflows/ci.yml:76]: I can try to extract those values from `MODULE.bazel` instead.

Comment (oharboe) [reply .github/workflows/ci.yml:76]: Sweet

Comment (oharboe) [reply .github/workflows/ci.yml:76]: Please do so in a followup PR. Merging for now so that pull requests that come in are tested.

**Pull request merged**